### DISCUSSION TITLE
Bundler gem installation mechanism for Bundler buildpack

### DISCUSSION THREAD

**Discussion by leondmello** [paketo-buildpacks/feedback#41] [Q&A]
Why does the bundler buildpack install pre-built bundler binaries instead of using `gem` to install the appropriate version?

**Comment by sophiewigmore** [comment]
I'm not sure if that was ever considered in the original implementation, cc @ryanmoran incase he has historical context.
That seems like a viable option! I think that the way the buildpack installs the binary now is more consistent with how we install tools across the project and collect metadata.

**Reply by ryanmoran** [reply]
The historical context is that we want to make sure the buildpack can be made to support offline platforms. If the buildpack uses `gem` then its going to need a network connection to fetch the `bundler` gem.

**Reply by robdimsdale** [reply]
I'm not familiar enough with the `gem install` process to know if we could leverage it in an offline environment. If we could do that, it could be worth exploring.

**Reply by fg-j** [reply]
Since I'm working on moving bundler dependency management out of the dep server now, I can investigate briefly whether it'd be possible to use `gem install` for bundler offline.

**Reply by ryanmoran** [reply]
The biggest requirement to meet would be that we want `bundler` installed in its own layer, not into the `gems` directory under the `mri` layer, which is what `gem install` would likely do unless you override its configuration.
